assets/js/lesson_22/to_do_list.js

```diff
@@ -0,0 +1,229 @@
+let ul = document.querySelector('#ul');
+let lastSelected = null;
+
+ul.addEventListener('click', function (e) {
+    if (e.target === this) {
+        return false;
+    }
+
+    if (!e.ctrlKey) {
+        clearSelected(this.children);
+    } else if (e.target.classList.contains('selected')) {
+        e.target.classList.remove('selected');
+        return;
+    }
+
+    if (e.shiftKey) {
+        selectRange(lastSelected, e.target);
+    } else {
+        addSelected(e.target);
+        lastSelected = e.target;
+    }
+})
```

Comment:
you have strange structure of this function. First you have logic of exiting from function - it's ok. But then you have a conditional branching, which really hard to read. 
you need to do something like:
```javascript
if (e.ctrlKey) {
 // ctrlKey behaviour 
} else if (e.shiftKey) {
 // shiftKey behaviour
} else {
   // any other behaviour
}
 ```

Reply:
done